From 3d5740c05b3f12096595522013473177836a86fc Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Wed, 20 Mar 2019 12:57:09 +0800
Subject: [PATCH 610/706] switch: tsn: set IFH_INJ_POP_CNT_DISABLE as SDK iset
 TSN driver to be configurable

It seems IFH_INJ_POP_CNT_DISABLE is necessary according to SDK, so set it.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/mscc/ocelot.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/mscc/ocelot.c b/drivers/net/ethernet/mscc/ocelot.c
index 1ca6e4aa04fc..769c5ca47212 100644
--- a/drivers/net/ethernet/mscc/ocelot.c
+++ b/drivers/net/ethernet/mscc/ocelot.c
@@ -527,9 +527,10 @@ static int ocelot_port_stop(struct net_device *dev)
 static int ocelot_gen_ifh(u32 *ifh, struct frame_info *info)
 {
 	ifh[0] = IFH_INJ_BYPASS;
-	ifh[1] = (0xf00 & info->port) >> 8;
+	ifh[1] = (0xff00 & info->port) >> 8;
 	ifh[2] = (0xff & info->port) << 24;
-	ifh[3] = (info->tag_type << 16) | info->vid;
+	ifh[3] = IFH_INJ_POP_CNT_DISABLE | (info->cpuq << 20) |
+		 (info->tag_type << 16) | info->vid;
 
 	return 0;
 }
-- 
2.17.1

